composition-js/src/merging/merge.ts

```diff
@@ -884,15 +893,175 @@ class Merger {
     return filtered;
   }
 
-  private hasExternal(sources: (FieldDefinition<any> | undefined)[]): boolean {
+  private hasExternal(sources: FieldOrUndefinedArray): boolean {
     return sources.some((s, i) => s !== undefined && this.isExternal(i, s));
   }
 
   private isShareable(sourceIdx: number, field: FieldDefinition<any>): boolean {
     return this.metadata(sourceIdx).isFieldShareable(field);
   }
 
-  private mergeField(sources: (FieldDefinition<any> | undefined)[], dest: FieldDefinition<any>) {
+  private getOverrideDirective(sourceIdx: number, field: FieldDefinition<any>): Directive<any> | undefined {
+    // Check the directive on the field, then on the enclosing type.
+    const metadata = this.metadata(sourceIdx);
+    const overrideDirective = metadata.isFed2Schema() ? metadata.overrideDirective() : undefined;
+    const allFieldOverrides = overrideDirective ? field.appliedDirectivesOf(overrideDirective) : [];
+    return allFieldOverrides[0]; // if array is empty, will return undefined
+  }
+
+  private overrideConflictsWithOtherDirective({
+    idx,
+    field,
+    subgraphName,
+    fromIdx,
+    fromField,
+    fromSubgraphName
+  }: {
+    idx: number;
+    field: FieldDefinition<any> | undefined;
+    subgraphName: string;
+    fromIdx: number;
+    fromField: FieldDefinition<any> | undefined;
+    fromSubgraphName: string;
+  }): { result: boolean, conflictingDirective?: DirectiveDefinition, subgraph?: string } {
+    const fromMetadata = this.metadata(fromIdx);
+    for (const directive of [fromMetadata.requiresDirective(), fromMetadata.providesDirective()]) {
+      if (fromField?.hasAppliedDirective(directive)) {
+         return {
+           result: true,
+           conflictingDirective: directive,
+           subgraph: subgraphName,
+         };
+       }
+    }
+    if (field && this.isExternal(idx, field)) {
+      return {
+        result: true,
+        conflictingDirective: fromMetadata.externalDirective(),
+        subgraph: subgraphName,
+      };
+    }
+    if (fromField && this.isExternal(fromIdx, fromField)) {
```

Comment:
This actually only just occurred to me, but I think we should remove that condition. The reason is that if an overridden field is still "used" in the from subgraph, then instead of removing the field, you may want to mark it `@external` instead. But the process you would then want is:
1. override some field `f` from subgraph A in subgraph B
2. mark `f` in subgraph A as `@external` explicitly.
3. now remove the `@override` in subgraph B.

Except that currently, this rule will prevent step 2.

Overall, overriding a field that is marked external is not that different from adding an overriding point to a subgraph that don't have the field at all, and we don't error out in that later case, so it make sense not too in the former either.

Reply:
ok, that makes sense.